<launch>
    <arg name="base_frame_id" value="turtle1"/>
    <arg name="fixed_frame_id" value="world"/>

    <node pkg="turtlesim" type="turtlesim_node" name="turtle1"/>
    <node pkg="turtle_tf" type="turtle_tf_broadcaster" name="turtle1_tf" args="turtle1"/>

    <node pkg="wifi_mapping" type="simulated_ap_node.py" name="ap_1">
        <param name="base_frame_id" value="$(arg base_frame_id)"/>
        <param name="fixed_frame_id" value="$(arg fixed_frame_id)"/>
        <param name="model_type" value="log_distance"/>
        <param name="model_params" value='{"location": [5,5,0], "transmit_power": 30, "path_loss_exponent": 0.5}'/>
    </node>
    <node pkg="wifi_mapping" type="simulated_ap_node.py" name="ap_2">
        <param name="base_frame_id" value="$(arg base_frame_id)"/>
        <param name="fixed_frame_id" value="$(arg fixed_frame_id)"/>
        <param name="model_type" value="log_distance"/>
        <param name="model_params" value='{"location": [0,10,3], "transmit_power": 20, "path_loss_exponent": 1.0}'/>
    </node>
    <node pkg="wifi_mapping" type="simulated_ap_node.py" name="ap_3">
        <param name="base_frame_id" value="$(arg base_frame_id)"/>
        <param name="fixed_frame_id" value="$(arg fixed_frame_id)"/>
        <param name="model_type" value="log_distance"/>
        <param name="model_params" value='{"location": [10,1,0], "transmit_power": 20, "path_loss_exponent": 1.0}'/>
    </node>

    <!-- node pkg="wifi_mapping" type="wifi_mapping_node" name="wifi_gp" output="screen" launch-prefix="gdb -ex run args" -->
    <node pkg="wifi_mapping" type="wifi_mapping_node" name="wifi_gp" output="screen">
        <param name="wifi_topic" value="/wifi_measurement"/>
        <param name="base_frame_id" value="$(arg base_frame_id)"/>
        <param name="fixed_frame_id" value="$(arg fixed_frame_id)"/>
    </node>
</launch>
